frontend/src/views/Game.tsx

```diff
@@ -84,6 +85,22 @@ function GamePage() {
     setTimeUp(newGame.gameTimer.timeUp);
   };
 
+  const setDefaultCodeFromProblems = useCallback((problemsParam: Problem[]) => {
+    const promises: Promise<DefaultCodeType>[] = [];
+    problemsParam.forEach((problem) => {
+      if (problem && problem.problemId) {
+        promises.push(getDefaultCodeMap(problem.problemId));
+      }
+    });
+
+    // Get the result of promises and set the default code list.
+    Promise.all(promises).then((result) => {
+      setDefaultCodeList(result);
+    }).catch((err) => {
+      setError(err.message);
+    });
```

Comment:
If I'm understanding correctly, this waits for all the default code maps for each language to be returned and then the result object is a list of the code maps?

Reply:
This waits for the default code maps (which have the code for each language) _for each **Problem**_ in the Game object to be returned. Yes, the result is a list of code maps, where each code map corresponds to a different Problem and the code maps have the language as the key and the String of start code as the value.